src/async-wrap.cc

```diff
@@ -351,6 +404,23 @@ static void SetupHooks(const FunctionCallbackInfo<Value>& args) {
   SET_HOOK_FN(destroy);
   env->AddPromiseHook(PromiseHook, nullptr);
 #undef SET_HOOK_FN
+
+  {
+    Local<FunctionTemplate> ctor =
+        FunctionTemplate::New(env->isolate());
+    ctor->SetClassName(FIXED_ONE_BYTE_STRING(env->isolate(), "PromiseWrap"));
+    Local<ObjectTemplate> promise_wrap_template = ctor->InstanceTemplate();
+    promise_wrap_template->SetInternalFieldCount(
+        PromiseWrap::kInternalFieldCount);
+    promise_wrap_template->SetAccessor(
+        FIXED_ONE_BYTE_STRING(env->isolate(), "promise"),
+        PromiseWrap::GetPromise);
+    promise_wrap_template->SetAccessor(
+        FIXED_ONE_BYTE_STRING(env->isolate(), "parentId"),
+        PromiseWrap::GetParentId);
+    env->SetProtoMethod(ctor, "getAsyncId", AsyncWrap::GetAsyncId);
```

Comment:
I thought it’s nice to have for consistency, but removing it is trivial. I’ve removed it for now, we can always revisit if necessary.

I haven’t measured, but I can’t really imagine it’s costly – it adds a method to the prototype, which *should* be practically free.